acceptance/tests/loader/autoload_from_resource_type_decl.rb

```diff
@@ -4,80 +4,64 @@
   require 'puppet/acceptance/environment_utils.rb'
   extend Puppet::Acceptance::EnvironmentUtils
 
-  require 'master_manipulator'
-  extend MasterManipulator::Log
-
-  puppet_server_log      = puppet_server_log_path(master)
   app_type               = File.basename(__FILE__, '.*')
   tmp_environment        = mk_tmp_environment_with_teardown(master, app_type)
   fq_tmp_environmentpath = "#{environmentpath}/#{tmp_environment}"
 
   relative_type_dir    = 'modules/one/lib/puppet/type'
-  relative_type_1_path = "#{relative_type_dir}/type_tst.rb"
+  relative_type_path   = "#{relative_type_dir}/type_tst.rb"
+  relative_helper_dir  = "#{relative_type_dir}/helper"
+  relative_helper_path = "#{relative_helper_dir}/required_helper.rb"
   step 'create custom type' do
-    on(master, "mkdir -p #{fq_tmp_environmentpath}/#{relative_type_dir}")
+    on(master, "mkdir -p '#{fq_tmp_environmentpath}/#{relative_type_dir}' '#{fq_tmp_environmentpath}/#{relative_helper_dir}'")
 
     custom_type = <<-END
+    require 'puppet/type/helper/required_helper'
     Puppet::Type.newtype(:type_tst) do
       newparam(:name, :namevar => true) do
         Puppet.notice("found_type_tst")
       end
     end
     END
-    create_remote_file(master, "#{fq_tmp_environmentpath}/#{relative_type_1_path}", custom_type)
+    create_remote_file(master, "#{fq_tmp_environmentpath}/#{relative_type_path}", custom_type)
+
+    helper_content = <<-END
+    # this is an empty file
+    END
+    create_remote_file(master, "#{fq_tmp_environmentpath}/#{relative_helper_path}", helper_content)
 
     site_pp = <<-PP
     Resource['type_tst'] { 'found_type': }
     PP
     create_sitepp(master, tmp_environment, site_pp)
   end
 
-  on(master, "chmod -R 755 /tmp/#{tmp_environment}")
-
-  # rotate the server log so that it won't rotate during the test
-  rotate_puppet_server_log(master)
+  on(master, "chmod -R 755 '/tmp/#{tmp_environment}'")
 
+  # This will fail because the agent side will not be able to find the "required_helper.rb"
```

Comment:
You might want to assert something in the agent's stdout, about how compilation failed. I'm not sure what information the server returns in that case other than HTTP 500